core/src/main/java/tachyon/master/BlockInfo.java

```diff
@@ -76,8 +78,15 @@ public static int computeInodeId(long blockId) {
   public final long mOffset;
   public final long mLength;
 
+  /**
+   * Maps workerIds to the worker address they refer to, for workers that have cached this block
+   */
   private final Map<Long, NetAddress> mLocations = new HashMap<Long, NetAddress>(5);
-  private final Map<NetAddress, Long> mStorageDirIds = new HashMap<NetAddress, Long>(5);
+
+  /**
+   * Maps workerIds to the storageIds they have pages cached in.
+   */
+  private final Map<Long, Set<Long>> mWorkerDirs = new HashMap<Long, Set<Long>>();
```

Comment:
Will operations on the two HashMaps be threadsafe? For example if `removeLocation` and `addLocation` are called from different threads?